src/Services/LightningService.cs

```diff
@@ -250,8 +248,20 @@ public async Task OpenChannel(ChannelOperationRequest channelOperationRequest)
                     throw new InvalidOperationException();
                 }
 
+                // Check features to see if we need or is allowed to add a close address
+                var upfrontShutdownScriptOpt = remoteNodeInfo.Features.ContainsKey((uint)FeatureBit.UpfrontShutdownScriptOpt);
+                var upfrontShutdownScriptReq = remoteNodeInfo.Features.ContainsKey((uint)FeatureBit.UpfrontShutdownScriptReq);
+                string? closeAddress = null;
+                if (upfrontShutdownScriptOpt && remoteNodeInfo.Features[(uint)FeatureBit.UpfrontShutdownScriptOpt] is { IsKnown: true } ||
+                    upfrontShutdownScriptReq && remoteNodeInfo.Features[(uint)FeatureBit.UpfrontShutdownScriptReq] is { IsKnown: true })
+                {
+                    var keyPathInformation = await GetCloseAddress(channelOperationRequest, derivationStrategyBase, _nbXplorerService, _logger);
+                    closeAddress = keyPathInformation?.Address?.ToString();
```

Comment:
Maybe here check that it is not null the closeAddress?

Reply:
`GetCloseAddress` throws if the close address is null

Reply:
`GetCloseAddress` throws if there is no close address. The signature of the function is wrong, it should return Task<KeyPathInformation> instead of Task<KeyPathInformation?>